### PR TITLE
R4R: Add the threshold into SoftwareUpgradeProposal

### PR DESCRIPTION
**Pull request by MrXJC**

### CODE REVIEW

**Comment by haifengxi** [comment]
Should we incorporate this enhancement into v0.10 as well?  @wukongcheng

File: client/gov/cli/sendtx.go [hunk 1]
```diff
@@ -103,6 +108,7 @@ func GetCmdSubmitProposal(cdc *codec.Codec) *cobra.Command {
 	cmd.Flags().String(flagVersion, "0", "the version of the new protocol")
 	cmd.Flags().String(flagSoftware, " ", "the software of the new protocol")
 	cmd.Flags().String(flagSwitchHeight, "0", "the switchheight of the new protocol")
+	cmd.Flags().String(flagThreshold, "0.667", "the threshold of the software upgrade")
```
Comment (wukongcheng) [inline client/gov/cli/sendtx.go]: the upgrade signal threshold of the software upgrade

File: modules/upgrade/handler.go [hunk 2]
```diff
@@ -16,8 +16,11 @@ func EndBlocker(ctx sdk.Context, uk Keeper) (tags sdk.Tags) {
 			uk.DeleteSignal(ctx, upgradeConfig.Protocol.Version, (sdk.ConsAddress)(ctx.BlockHeader().ProposerAddress).String())
 		}
 
+		x := upgradeConfig.Threshold.String()
+		_ = x
```
Comment (chengwenxi) [inline modules/upgrade/handler.go]: remove

Comment (MrXJC) [reply modules/upgrade/handler.go]: OK👌

File: modules/gov/msgs.go [hunk 3]
```diff
@@ -124,6 +126,11 @@ func (msg MsgSubmitSoftwareUpgradeProposal) ValidateBasic() sdk.Error {
 	if err != nil {
 		return err
 	}
+	// if threshold not in [0.85,1), then print error
+	if msg.Threshold.LT(sdk.NewDecWithPrec(85,2)) || msg.Threshold.GTE(sdk.NewDec(1)){
```
Comment (wukongcheng) [inline modules/gov/msgs.go]: Why threshold must >= 85% but the flag threshold's default value is 0.667 ?

File: modules/upgrade/handler.go (hunk 2 above)
Comment (wukongcheng) [inline modules/upgrade/handler.go]: remove the debug code

**Comment by wukongcheng** [comment]
> Should we incorporate this enhancement into v0.10 as well? @wukongcheng

No.  v0.10 only accepts the bug fix but not any other user interface changes.

File: modules/gov/msgs.go (hunk 3 above)
Comment (haifengxi) [reply modules/gov/msgs.go]: Right, it doesn't make sense to have a default value of 0.667 while the valid range is [0.85,1) @MrXJC

Comment (haifengxi) [reply modules/gov/msgs.go]: Update related docs as well: 0.667 --> 0.85 @MrXJC

**Pull request merged**